### DISCUSSION TITLE
Conflicting dependencies in v1.10.15 constraints

### DISCUSSION THREAD

**Discussion by alexpbook** [apache/airflow#18775] [Q&A] [answered]
Hello all,

Installing apache-airflow[gcp] == 1.10.15, and using the constrains file:
https://raw.githubusercontent.com/apache/airflow/constraints-1.10.15/constraints-3.6.txt

Fails with the following error:
```
ERROR: Cannot install google-cloud-bigquery[bqstorage,pandas]==2.4.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    google-cloud-bigquery[bqstorage,pandas] 2.4.0 depends on pyarrow<3.0dev and >=1.0.0; extra == "pandas"
    The user requested (constraint) pyarrow==0.17.1
```

I don't think the constraint for pyarrow is necessary, and I see there was a PR to remove it, but it's still there:
https://github.com/apache/airflow/pull/12683

Can that line be removed from the constraints?
Is there any workaround for this?

**Comment by potiuk** [comment, marked as answer]
Since Airflow 1.10 is end-of-life, We are not going to update the constraints for 1.10. The workaround is to copy the constraints locally + modifying what's needed. However I believe it should work if you use old version of PIP as instructed in the instructions: 

https://airflow.apache.org/docs/apache-airflow/1.10.15/installation.html

if you look at the notes, you should use `pip` 20.2.4 to install airflow 1.10.15

```
On November 2020, new version of PIP (20.3) has been released with a new, 2020 resolver. This resolver does not yet work with Apache
Airflow and might leads to errors in installation - depends on your choice of extras. In order to install Airflow you need to either downgrade
pip to version 20.2.4 pip upgrade --pip==20.2.4 or, in case you use Pip 20.3, you need to add option --use-deprecated legacy-resolver to
your pip install command.
```

**Reply by alexpbook** [reply]
thanks, the workaround of downgrading PIP and using the legacy resolver worked.
I understand 1.10 is old, but we use airflow on GCP and unfortunately they don't offer v2 in a stable way yet.

**Reply by potiuk** [reply]
Actually as of a few weeks they do - Airflow 2 is in GA there.

**Reply by potiuk** [reply]
https://cloud.google.com/composer/docs/release-notes#September_15_2021
